packages/rainbowkit/src/wallets/walletConnectors/desigWallet/desigWallet.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import type { InjectedConnectorOptions } from '@wagmi/core/connectors/injected';
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { Wallet } from '../../Wallet';
+
+export interface DesigWalletOptions {
+  chains: Chain[];
+}
+
+export const desigWallet = ({
+  chains,
+  ...options
+}: DesigWalletOptions & InjectedConnectorOptions): Wallet => {
+  return {
+    id: 'desig',
+    name: 'Desig Wallet',
+    iconUrl: async () => (await import('./desigWallet.svg')).default,
+    iconBackground: '#ffffff',
+    installed: (window as any)?.desig?.ethereum,
+    downloadUrls: {
+      android: 'https://play.google.com/store/apps/details?id=io.desig.app',
+      ios: 'https://apps.apple.com/app/desig-wallet/id6450106028',
+      qrCode: 'https://desig.io/download',
+      mobile: 'https://desig.io/download',
+      browserExtension:
+        'https://chrome.google.com/webstore/detail/desig-wallet/panpgppehdchfphcigocleabcmcgfoca',
+    },
+
+    createConnector: () => {
+      const getProvider = () =>
+        typeof window !== 'undefined'
+          ? (window as any).desig?.ethereum
```

Comment:
I'd also recommend you add support to Wagmi for your provider [here](https://github.com/wagmi-dev/references/blob/cf1b48dabbe42a9b71a4965928e5e789b6c656a0/packages/connectors/src/types.ts#L33) and [here](https://github.com/wagmi-dev/references/blob/cf1b48dabbe42a9b71a4965928e5e789b6c656a0/packages/connectors/src/utils/getInjectedName.ts#L47). We rely on Wagmi's internal wallet identifiers as a fallback

Reply:
Thank you for your prompt feedback and I apologize for my delay. I have updated the logo to 28x28 and added support for Wagmi to my provider. You can check it out in my PR [here](https://github.com/wagmi-dev/references/pull/425.) Please let me know if there are any further issues. Thank you very much